src/main/java/org/truffleruby/core/array/ArrayNodes.java

```diff
@@ -1876,13 +1876,15 @@ public int size(DynamicObject array,
     public abstract static class SortNode extends ArrayCoreMethodNode {
 
         @Child private CallDispatchHeadNode compareDispatchNode = CallDispatchHeadNode.create();
-        @Child private YieldNode yieldNode = new YieldNode();
         @Child private FixnumLowerNode fixnumLowerNode = FixnumLowerNode.create();
 
         private final BranchProfile errorProfile = BranchProfile.create();
 
+        protected final Assumption fixnumCmpAssumption = getContext().getCoreMethods().fixnumCmpAssumption;
+        protected final Assumption floatCmpAssumption = getContext().getCoreMethods().floatCmpAssumption;
```

Comment:
Why not have these expressions in the `assumptions` annotation directly?
We usually don't rely on having the RubyContext available for fields I think, nor on node constructors.

Reply:
It's a long (and final) expression I didn't want in a String I can't easily refactor. But, switching it gets past the NPE I'm seeing, so that's a good reason to change.